src/HTTPure/Body.purs

```diff
@@ -17,45 +17,67 @@ import Node.Encoding as Encoding
 import Node.HTTP as HTTP
 import Node.Stream as Stream
 
+import HTTPure.Headers as Headers
+
+newtype Chunked = Chunked (Stream.Readable ())
```

Comment:
We have to use a newtype here, because typeclass instances don't support row definitions in their heads, and `Stream.Readable` is just a type synonym for `Stream.Stream (read :: Stream.Read | s)`.  I don't know that purescript provides any workaround, I've heard tell that there's a `~` operator that can be used to get around this but I'm not sure if it's actually landed and haven't been able to find any docs.

Reply:
I was just thinking, as another approach, instead of this, we work towards the API you mentioned in https://github.com/cprussin/purescript-httpure/pull/104#issuecomment-416037317 and we could make a typeclass instance for something like this:

```purescript
(String -> Effect.Effect Unit) -> Aff.Aff Unit
```

That would lead to an API that looks more like what you mentioned:

```purescript
router _ = HTTPure.ok \writeData -> do
    writeData "hello"
    writeData "world"
```

I'd want to figure out some clean API around sending strings or binary data with this if we go this route.

Reply:
... and it might be best to leave defining the callback API for later when there's actually a real use case for it. At least I don't have a use case in mind at the moment, the only thing I need is sane serving of static files.

Reply:
Yeah, I agree, both is good and I can revisit the callback API later.  I would like to implement it before we release 1.0, but I'm fine with doing another minor release without it in the meantime.

Also, per https://github.com/purescript/purescript/issues/1510, there is after all a way to declare the typeclass for the stream without the newtype wrapper, so I'll get that change done in this PR.